#! /bin/sh -e

# DP: Add Debian URL for bug reporting isntructions.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir=$3/
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
	patch $pdir -f --no-backup-if-mismatch -p0 < $0
	if grep -q debian_bug_report_url ${dir}gcc/version.c; then
	  :
	else
	  echo 'const char debian_bug_report_url[] = "<URL:file:///usr/share/doc/gcc-4.0/README.Bugs>";' >> ${dir}gcc/version.c
	fi
	;;
    -unpatch)
	grep -v debian_bug_report_url ${dir}gcc/version.c > ${dir}gcc/version.c.new
	${dir}move-if-change ${dir}gcc/version.c.new ${dir}gcc/version.c
	patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
	;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

diff -u gcc/java/gjavah.c.orig gcc/java/gjavah.c
--- gcc/java/gjavah.c.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/java/gjavah.c	2004-10-09 18:13:09.000000000 +0200
@@ -2316,6 +2316,8 @@
   printf ("\n");
   printf (_("For bug reporting instructions, please see:\n"
 	    "%s.\n"), bug_report_url);
+  printf (_("For Debian GNU/Linux specific bug reporting instructions, please see:\n"
+	    "%s.\n"), debian_bug_report_url);
   exit (0);
 }
 
diff -u gcc/java/jcf-dump.c.orig gcc/java/jcf-dump.c
--- gcc/java/jcf-dump.c.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/java/jcf-dump.c	2004-10-09 18:13:09.000000000 +0200
@@ -906,6 +906,8 @@
   printf ("\n");
   printf (_("For bug reporting instructions, please see:\n"
 	    "%s.\n"), bug_report_url);
+  printf (_("For Debian GNU/Linux specific bug reporting instructions, please see:\n"
+	    "%s.\n"), debian_bug_report_url);
   exit (0);
 }
 
diff -u gcc/java/jv-scan.c.orig gcc/java/jv-scan.c
--- gcc/java/jv-scan.c.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/java/jv-scan.c	2004-10-09 18:13:09.000000000 +0200
@@ -118,6 +118,8 @@
   printf ("\n");
   printf (_("For bug reporting instructions, please see:\n"
 	    "%s.\n"), bug_report_url);
+  printf (_("For Debian GNU/Linux specific bug reporting instructions, please see:\n"
+	    "%s.\n"), debian_bug_report_url);
   exit (0);
 }
 
diff -u gcc/diagnostic.c.orig gcc/diagnostic.c
--- gcc/diagnostic.c.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/diagnostic.c	2004-10-09 18:15:43.000000000 +0200
@@ -237,6 +237,8 @@
       fnotice (stderr, "Please submit a full bug report,\n"
 	       "with preprocessed source if appropriate.\n"
 	       "See %s for instructions.\n", bug_report_url);
+      fnotice (stderr, "For Debian GNU/Linux specific bug reporting instructions,\n"
+	       "see %s.\n", debian_bug_report_url);
       exit (FATAL_EXIT_CODE);
 
     case DK_FATAL:
diff -u gcc/gcc.c.orig gcc/gcc.c
--- gcc/gcc.c.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/gcc.c	2004-10-09 18:13:09.000000000 +0200
@@ -2883,9 +2883,11 @@
 		    fatal ("\
 Internal error: %s (program %s)\n\
 Please submit a full bug report.\n\
-See %s for instructions.",
+See %s for instructions.\n\
+For Debian GNU/Linux specific bug reporting instructions, see\n\
+%s.\n",
 			   strsignal (WTERMSIG (status)), commands[j].prog,
-			   bug_report_url);
+			   bug_report_url, debian_bug_report_url);
 		  signal_count++;
 		  ret_code = -1;
 		}
@@ -6336,6 +6338,8 @@
 	{
 	  printf (_("\nFor bug reporting instructions, please see:\n"));
 	  printf ("%s.\n", bug_report_url);
+	  printf ("For Debian GNU/Linux specific bug reporting instructions, please see:\n");
+	  printf ("%s.\n", debian_bug_report_url);
 
 	  return (0);
 	}
@@ -6615,6 +6619,8 @@
     {
       printf (("\nFor bug reporting instructions, please see:\n"));
       printf ("%s\n", bug_report_url);
+      printf ("For Debian GNU/Linux specific bug reporting instructions, please see:\n");
+      printf ("%s\n", debian_bug_report_url);
     }
 
   return (signal_count != 0 ? 2
diff -u gcc/gccbug.in.orig gcc/gccbug.in
--- gcc/gccbug.in.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/gccbug.in	2004-10-09 18:13:09.000000000 +0200
@@ -25,13 +25,13 @@
 VERSION=3.113
 
 # The submitter-id for your site.
-SUBMITTER=net
+SUBMITTER=net # net-debian
 
 # The default mail address for PR submissions. 
-GNATS_ADDR=gcc-gnats@gcc.gnu.org
+GNATS_ADDR="gcc-gnats@gcc.gnu.org,debian-gcc@lists.debian.org"
 
 # The default release for this host.
-DEFAULT_RELEASE="@gcc_version_full@"
+DEFAULT_RELEASE="@gcc_version_full@ (Debian testing/unstable)"
 
 # The default organization.
 DEFAULT_ORGANIZATION=
diff -u gcc/gcov.c.orig gcc/gcov.c
--- gcc/gcov.c.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/gcov.c	2004-10-09 18:13:09.000000000 +0200
@@ -399,6 +399,8 @@
   fnotice (file, "  -u, --unconditional-branches    Show unconditional branch counts too\n");
   fnotice (file, "\nFor bug reporting instructions, please see:\n%s.\n",
 	   bug_report_url);
+  fnotice (file, "\nFor Debian GNU/Linux specific bug reporting instructions, please see:\n%s.\n",
+	   debian_bug_report_url);
   exit (status);
 }
 
diff -u gcc/version.h.orig gcc/version.h
--- gcc/version.h.orig	2004-10-09 18:12:39.000000000 +0200
+++ gcc/version.h	2004-10-09 18:13:09.000000000 +0200
@@ -2,4 +2,5 @@
 #define GCC_VERSION_H
 extern const char version_string[];
 extern const char bug_report_url[];
+extern const char debian_bug_report_url[];
 #endif /* ! GCC_VERSION_H */
